File: examples/dart/lib/gcs_context.dart
import 'package:mavlink/mavlink_protocols.dart';

/// Ground control station identity (MAVLink convention).
const gcsSystemId = 255;
const gcsComponentId = 190;

/// Shared MAVLink GCS state for the interactive SITL example.
class GcsContext {
  GcsContext({
    required this.gcs,
    required this.vehicle,
    required this.client,
  });

  final MavlinkGcs gcs;
  final MavlinkNode vehicle;
  final MavlinkVehicleClient client;

  /// Cancels in-flight parameter/mission operations (type `cancel` in CLI).
  MavlinkCancellationToken? operationCancel;

  MavlinkSession get session => gcs.session;
  HeartbeatMonitor get heartbeatMonitor => gcs.heartbeatMonitor;
  HeartbeatPublisher get heartbeatPublisher => gcs.heartbeatPublisher;

  ParameterProtocol get parameters => client.parameters;
  MissionProtocol get mission => client.mission;
  CommandProtocol get command => client.command;

  int get targetSystem => vehicle.systemId;
  int get targetComponent => vehicle.componentId;
}

File: examples/csharp/GcsContext.cs
using Mavlink;

namespace MavlinkSitlGcs;

/// <summary>Ground control station identity (MAVLink convention).</summary>
public static class GcsIdentity
{
    public const byte SystemId = 255;
    public const byte ComponentId = 190;
}

/// <summary>Shared MAVLink GCS state for the interactive SITL example.</summary>
public sealed class GcsContext
{
    public GcsContext(MavlinkGcs gcs, MavlinkNode vehicle, MavlinkVehicleClient client)
    {
        Gcs = gcs;
        Vehicle = vehicle;
        Client = client;
    }

    public MavlinkGcs Gcs { get; }

    public MavlinkNode Vehicle { get; }

    public MavlinkVehicleClient Client { get; }

    /// <summary>Cancels in-flight parameter/mission operations (type <c>cancel</c> in CLI).</summary>
    public MavlinkCancellationToken? OperationCancel { get; set; }

    public MavlinkSession Session => Gcs.Session;

    public HeartbeatMonitor HeartbeatMonitor => Gcs.HeartbeatMonitor;

    public HeartbeatPublisher HeartbeatPublisher => Gcs.HeartbeatPublisher;

    public ParameterProtocol Parameters => Client.Parameters;

    public MissionProtocol Mission => Client.Mission;

    public CommandProtocol Command => Client.Command;

    public byte TargetSystem => Vehicle.SystemId;

    public byte TargetComponent => Vehicle.ComponentId;
}
